tests/test.py

```diff
@@ -144,6 +144,28 @@ def test_ctor(self):
             'Version parameter must be a string or None. Found float'
         )
 
+    def test_url_valid_resource(self):
+        url = self.client._url('/hello/{0}/world', 'somename')
+        self.assertEqual(
+            url, '{0}{1}'.format(url_prefix, 'hello/somename/world')
+        )
+
+        url = self.client._url('/hello/{0}/world', '/some?name')
+        self.assertEqual(
+            url, '{0}{1}'.format(url_prefix, 'hello/%2Fsome%3Fname/world')
+        )
```

Comment:
FYI, you can just use `assert` with pytest:

```python
assert url == '{0}{1}'.format(url_prefix, 'hello/somename/world')
```

Assertion failures are still formatted nicely with helpful information etc. It's rad as heck.